### PR TITLE
fixes the ark healing itself

### PR DESCRIPTION
**Pull request by ChangelingRain**
wow woah

### CODE REVIEW

**Comment by ChangelingRain** [comment]
funnily enough this wouldn't have affected the round in question because the ark didn't get hit at all after it got bombed

File: code/game/gamemodes/clock_cult/clock_structure.dm [hunk 1]
```diff
@@ -29,7 +29,8 @@
 	return ..()
 
 /obj/structure/destructible/clockwork/ratvar_act()
-	obj_integrity = max_integrity
+	if(ratvar_awakens || clockwork_gateway_activated)
+		obj_integrity = max_integrity
```
Comment (Cyberboss) [inline code/game/gamemodes/clock_cult/clock_structure.dm]: If you want it to be indestructible when the rat is awake, you should probably block the base calls of obj_break and obj_destruction if he is.

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/clock_cult/clock_structure.dm (hunk 1 above)
Comment (ChangelingRain) [inline code/game/gamemodes/clock_cult/clock_structure.dm]: I don't, though, this is just getting healed by ratvar

**Comment by Robustin** [comment]
Maybe it wouldve affected the round because the ark was healing any damage taken prior to the bombing.

If I understand it though, and im just guessing here, but the Clockwork Gateway was doing Ratvar_Act (to spread the brass shit when it gets close) and in turn healing itself to full?

Also plz stop making your antags bomb-proof. I fear the day you start looking at wizard and your first change is to make them 99% resistant to ex_act.

**Comment by ChangelingRain** [comment]
It's not bomb-proof you just need three bombs or some followup. You know, a plan, rather than reckless explosions.

**Comment by PKPenguin321** [comment]
Kinda lame, should die to two bombs tops

**Comment by Robustin** [comment]
I think my biggest issue with the bomb resistance is that no player will EVER have that kind of knowledge. Bombs have been a one-and-done solution for removing local threats for years, multi-maxcap has never been a strategy for almost anything because it was just excessive (or futile, for blobs).

You say you want the attackers to have a "plan" but how the fuck is any non-coder supposed to know how to "plan" how many bombs to use? It's extremely unintuitive and if I hadn't ended up killing a dozen cultists/future converts then I might have inadvertently handed the cult a victory. Requiring people to "plan" an attack only makes sense if they have reasonable access to the information that would let them effectively plan things. I mean its kind of trivial for me to prepare 3 maxcaps instead of 1, the only thing that hindered me was that knowledge gap. Now I'll just be sure to bring 3 bombs and the explosions status quo continues for me but will fuck up anyone who's not reading the code.

I sympathize with your attitude on bombs, the biggest balancing factor on bombs is the obscure knowledge required to assemble them and the high stakes for failure (how many people are willing to risk a ban to deny antags greentext?). Neither of those are really "IC" balancing conditions. 

I think a great long term solution would be for bomb damage to structures/blobs/etc. to scale depending on the bomb size, and it would scale PAST THE MAXCAP. The station maxcap would still be 8/12/20 or whatever but a point blank 8/12/20 bomb would only do a fraction of the damage that a 20/44/88 or whatever max maxcap bomb would do. This would still serve as "fringe" knowledge at first, but it would eventually diffuse into the metagame as people learned and adapted and would serve as a better long-term framework for toxins balance rather than "You need *arbitrary and frequently changing number* of maxcaps to kill _______". A max maxcap takes a lot of effort and precision to make and should be the gold standard for "this will absolutely annihilate anything it detonates near, no exceptions".

**Comment by PKPenguin321** [comment]
Oldman_fucking_explodes.txt

**Comment by ChangelingRain** [comment]
So you're saying I should make it explicitly immune to bombs instead.

**Comment by Robustin** [comment]
What I said was that bomb damage should scale beyond the maxcap to give toxins a better balance framework and reward scientists who actually use it like a science department and not just a quick and dirty maxcap factory. 

But at the end of the day I'd take a bomb-immune ark that is reasonably assailable over the current metagame of "good luck reaching the ark I'm behind 8 ocular wardens backed up by an AI that can turret/foam/flash/bolt/beepsky along with the dozen other defensive memes (stun runes, brain damage/confusion spam, chain lightning, anima, guardians, proseltyzer) that makes bombs MANDATORY for assailing a competent cult satellite. 

Keep in mind we used to have a MODE where the ENTIRE CREW had to assail the AI on its satellite and the AI's only defense was its borgs and the ability to destroy a few machines. Endgame clockcult is that same mode PLUS all the cultists, constructs, holo... guardians, wardens, stunmines, etc. 

If you want the endgame to be healthier I would suggest what I was considering for Blood Cult, require the Clock Cult to summon in one of three randomly selected rooms. Or at least make it so that the AI isn't an autoconvert once someone figures out the insanely difficult (/s) combo of flash + converting a borg. 

Or alternatively make borg conversion require the tier 3 conversion rune. Either way I think the meta is still shaking out on Clock Cult but people are going to realize that all the early game tactics don't mean shit, all you need to do is get a borg alone (trivial with law 2) and then flash+convert it. Have the borg quickly spirit you in a locker to the AI core, from there the AI is helpless and now you've got a perma-subverted AI, quite possibly the greatest advantage any team antag can get on SS13, and no more worrying about borgs/security discovering your base unless the AI does something dumb enough to give itself away.

**Comment by ChangelingRain** [comment]
So you're saying converting the AI should make it obviously ratvar'd, instead of being unnoticeable.

**Pull request merged**